lib/python/picongpu/picmi/diagnostics/openpmd_sources/charge_density.py

```diff
@@ -0,0 +1,73 @@
+"""
+This file is part of PIConGPU.
+Copyright 2021-2025 PIConGPU contributors
+Authors: Masoud Afshari
+License: GPLv3+
+"""
+
+import typeguard
+from typing import Optional
+
+
+@typeguard.typechecked
+class ChargeDensity:
+    """
+    Represents the ChargeDensity data source for the openPMD plugin in PIConGPU.
+
+    This class defines the charge density field, derived from particle species at runtime,
+    which can be output using the openPMD plugin. An optional filter can be applied to select
+    which particles contribute to the charge density.
```

Comment:
same problem as before, we should not refer to PIConGPU specific things on the PICMI side